src/Features/VisualBasic/Portable/ConvertLinq/VisualBasicConvertLinqQueryToLinqMethodProvider.vb

```diff
@@ -0,0 +1,234 @@
+                    expressions.Select(Function(expression) SyntaxFactory.SimpleArgument(expression)))))
+                Dim memberAccessExpression = SyntaxFactory.MemberAccessExpression(
+                    SyntaxKind.SimpleMemberAccessExpression, parentExpression,
+                    SyntaxFactory.Token(SyntaxKind.DotToken),
+                    SyntaxFactory.IdentifierName(keyword))
+                Return SyntaxFactory.InvocationExpression(memberAccessExpression, argumentList)
+            End Function
+
+            Private Function ProcessOrderByClause(expression As ExpressionSyntax, orderByClause As OrderByClauseSyntax) As ExpressionSyntax
+                Dim isFirst = True
+                For Each ordering In orderByClause.Orderings
+                    Dim keyword As String
+                    Select Case (ordering.Kind())
+                        Case SyntaxKind.AscendingOrdering
+                            keyword = If(isFirst, NameOf(Enumerable.OrderBy), NameOf(Enumerable.ThenBy))
+                        Case SyntaxKind.DescendingOrdering
+                            keyword = If(isFirst, NameOf(Enumerable.OrderByDescending), NameOf(Enumerable.ThenByDescending))
+                        Case Else
+                            Return Nothing
+                    End Select
+                    expression = CreateInvocationExpression(expression, keyword, ordering.Expression)
+                    isFirst = False
+                Next
+
+                Return expression
+            End Function
+
+            Private Shared Function BeautifyName(name As String) As String
+                Return name.Replace("$", "")
+            End Function
+
+            Private Function MakeIdentifierNameReplacements(node As SyntaxNode) As VisualBasicSyntaxNode
+                Return DirectCast(New LambdaRewriter(Function(n) GetIdentifierName(n)).Visit(node), VisualBasicSyntaxNode)
+            End Function
+
+            Private Function GetIdentifierName(node As IdentifierNameSyntax) As String
+                Dim names = GetIdentifierNames(_semanticModel, node, _cancellationToken)
+                If (names.IsDefault) Then
+                    Return String.Empty
+                Else
+                    Return BeautifyName(String.Join(".", names))
+                End If
+            End Function
+
+            Private Class LambdaRewriter
+                Inherits VisualBasicSyntaxRewriter
+                Private _semanticModel As SemanticModel
+                Private _cancellationToken As CancellationToken
+                Private _getNameMethod As Func(Of IdentifierNameSyntax, String)
+
+                Public Sub New(getNameMethod As Func(Of IdentifierNameSyntax, String))
+                    _getNameMethod = getNameMethod
+                End Sub
+
+                Public Overrides Function VisitIdentifierName(node As IdentifierNameSyntax) As SyntaxNode
+                    Dim name = _getNameMethod(node)
```

Comment:
>Dim name = _getNameMethod(node) [](start = 20, length = 31)

There is really no need to go through all this trouble unless this node refers to a range variable. Also , the code should be defensive, it should ensure that it found what is expected.